spec/unit/hanami/events/adapter/redis_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'connection_pool'
+require 'redis'
+
+RSpec.describe Hanami::Events::Adapter::Redis do
+  let(:handler) { proc { |payload| payload } }
+  let(:adapter) { described_class.new(redis: redis) }
+
+  context '#subscribe' do
+    let(:redis) { ConnectionPool.new(size: 5, timeout: 5) { Redis.new } }
+
+    it 'pushes subscriber to the list of subscribers' do
+      expect {
+        adapter.subscribe('event.name', &handler)
+      }.to change { adapter.subscribers.count }.by(1)
+    end
+
+    it 'spawns just one thread' do
+      expect(Thread).to receive(:new).once
+
+      adapter.subscribe('user.created', &handler)
+      adapter.subscribe('user.updated', &handler)
+    end
+  end
+
+  context '#broadcast' do
+    let(:redis) { Redis.new }
+
+    it 'calls #broadcast method with proper params on redis' do
```

Comment:
Sorry, "proper params" is related to `#broadcast` call, not to redis. :)
Probably it's better to remove "on redis" part to make this description less ambiguous